tests/providers/microsoft/azure/hooks/test_adx.py

```diff
@@ -279,20 +279,28 @@ def test_run_query(self, mock_execute, mocked_connection):
         assert args[1]["properties"]._options["option1"] == "option_value"
 
     @pytest.mark.parametrize(
-        "mocked_connection",
+        "mocked_connection, warning",
         [
             pytest.param(
                 "a://usr:pw@host?extra__azure_data_explorer__tenant=my-tenant"
                 "&extra__azure_data_explorer__auth_method=AAD_APP",
+                True,
                 id="prefix",
             ),
-            pytest.param("a://usr:pw@host?tenant=my-tenant&auth_method=AAD_APP", id="no-prefix"),
+            pytest.param("a://usr:pw@host?tenant=my-tenant&auth_method=AAD_APP", False, id="no-prefix"),
         ],
-        indirect=True,
+        indirect=["mocked_connection"],
     )
-    def test_backcompat_prefix_works(self, mocked_connection):
+    def test_backcompat_prefix_works(self, mocked_connection, warning):
         hook = AzureDataExplorerHook(azure_data_explorer_conn_id=mocked_connection.conn_id)
-        assert hook.connection._kcsb.data_source == "host"
+        if warning:
+            with pytest.warns(
+                AirflowProviderDeprecationWarning,
+                match="`extra__azure_data_explorer__.*` is deprecated in azure connection extra, please use `.*` instead",
+            ):
+                assert hook.connection._kcsb.data_source == "host"
+        else:
+            assert hook.connection._kcsb.data_source == "host"
```

Comment:
I don't know if we need it?
https://github.com/apache/airflow/pull/27219/files suggets that that this was to preserve backward compatbility with min Airflow 2.5.0 now that we are on 2.7.0 maybe we can safely remove the backported code?
cc @dstandish

Reply:
Event I am not sure about it. @dstandish Can you confirm ^^?

Reply:
@eladkal Shall I go ahead and make changes as per your suggestion or wait for @dstandish inputs?

Reply:
If the code is for backport with airflow versions older than 2.7.0 then we can just remove it

Reply:
I have updated the code.